scripts/justmat/foulplay.lua

```diff
@@ -297,20 +304,10 @@ function redraw()
   end
   screen.stroke() 
   screen.update()
-end
 
-midi.add = function(dev)
-  print("foulplay: midi device added", dev.id, dev.name)
-  midi_device = dev
-end
-
-midi.remove = function(dev)
-  print("foulplay: midi device removed", dev.id, dev.name)
-  midi_device = nil
 end
 
-function cleanup()
-  midi.add = nil
-  midi.remove = nil
-  params:write("justmat/foulplay.pset")
+midi.add = function(dev)
+  dev.event = clk.process_midi
+  print("fairplay: midi device added", dev.id, dev.name)
```

Comment:
i haven't been following the latest midi changes but note that `hello_ack` is still checking for and assigning to `midi_device`.

https://github.com/monome/dust/blob/062fefec78f31b581c16158780c8f6e167735cb9/scripts/jah/hello_ack.lua#L286-L294

is that no longer the right thing to do?

/cc @antonhornquist

Reply:
Initially I had it written somewhat like your example, but when switching to BeatClock I followed Fairplays implementation. I just assumed it would be "the way".

Reply:
to be honest, i don't know that there is "the way" just yet so in the meantime having a few that will ultimately converge is probably the way to go.

thanks for following up!